src/SSHDebugPS/ConnectionManager.cs

```diff
@@ -153,5 +140,67 @@ public static bool ShowContainerPickerWindow(IntPtr hwnd, out string connectionS
             connectionString = string.Empty;
             return false;
         }
+
+        /// <summary>
+        /// Parses the SSH connection string. Expected format is some permutation of username@hostname:portnumber.
+        /// If not defined, will provide default values.
+        /// </summary>
+        internal static void ParseSSHConnectionString(string connectionString, out string userName, out string hostName, out int port)
+        {
+            userName = StringResources.UserName_PlaceHolder;
+            hostName = StringResources.HostName_PlaceHolder;
+            port = 22; // Default SSH port is 22
+
+            int atSignIndex = connectionString.IndexOf('@');
+            if (atSignIndex >= 0)
+            {
+                // Find if a username is specified
+                if (atSignIndex > 0)
+                {
+                    userName = connectionString.Substring(0, atSignIndex);
+                }
+
+                // Find the beginning of the hostname section. 
+                // Handle where the first character is '@'
+                int hostNameStartPos = atSignIndex + 1;
+                if (hostNameStartPos < connectionString.Length)
+                {
+                    hostName = connectionString.Substring(hostNameStartPos);
+                }
+                else
+                {
+                    // Using default hostName so we don't need to look for the port
+                    return;
+                }
+            }
+            else
+            {
+                hostName = connectionString;
+            }
+
+            // Find if a port is specified. Handle possible IPV6 by grabbing the last colon
+            int lastColonIndex = hostName.LastIndexOf(':');
```

Comment:
I think this is wrong - if I specify a connection string of "user@1:2:3:4:5:6:7:8", it returns host = "1:2:3:4:5:6:7" and port = "8".  I believe the only valid way to specify a port in an IPv6 address is to disambiguate by putting the entire address in square brackets, and the port afterwards, e.g. "[1:2:3:4:5:6:7:8]:22"

Reply:
Colon count: 
* 0, no port
* 1, has a port
* >2 ipv6, check for open and close `[]`